### PR TITLE
feat: add env to retain amplify app

### PR DESCRIPTION
**Pull request by garyleefight**
Tested locally

### CODE REVIEW

File: packages/amplify-cli/src/extensions/amplify-helpers/delete-project.ts [hunk 1]
```diff
@@ -62,7 +62,7 @@ export async function getConfirmation(context, env?) {
     return {
       proceed: true,
       deleteS3: true,
-      deleteAmpilfyApp: true,
+      deleteAmpilfyApp: !!!process.env.CLI_DEV_INTERNAL_DISABLE_AMPLIFY_APP_DELETION
```
Comment (edwardfoyle) [inline packages/amplify-cli/src/extensions/amplify-helpers/delete-project.ts:65]: nit no need for triple bang: https://stackoverflow.com/questions/21154510/the-use-of-the-triple-exclamation-mark

**Review by edwardfoyle — APPROVED** [review]

File: packages/amplify-cli/src/extensions/amplify-helpers/delete-project.ts [hunk 2]
```diff
@@ -62,7 +62,7 @@ export async function getConfirmation(context, env?) {
     return {
       proceed: true,
       deleteS3: true,
-      deleteAmpilfyApp: true,
+      deleteAmpilfyApp: !process.env.CLI_DEV_INTERNAL_DISABLE_AMPLIFY_APP_DELETION
```
Comment (kaustavghosh06) [inline packages/amplify-cli/src/extensions/amplify-helpers/delete-project.ts:65]: There is a deleteAmpilfyApp: true down below as well - do you need to modify that as well?

Comment (garyleefight) [reply packages/amplify-cli/src/extensions/amplify-helpers/delete-project.ts:65]: No need to. Tested already

File: packages/amplify-cli/src/extensions/amplify-helpers/delete-project.ts (hunk 1 above)
Comment (karthik2872) [reply packages/amplify-cli/src/extensions/amplify-helpers/delete-project.ts]: Folks, I installed 4.32.1 today and hitting an issue when doing amplify push ( after amplify init )

What is this initialization, I see the cli.json file in my amplify folder , no issues there ( must have been created at init time )
How do I overcome this issue to continue to do amplify push. Please share your thoughts ASAP as I am on a tight deadline
**********************************************
FeatureFlags is not initialized
An error occurred during the push operation: FeatureFlags is not initialized
***************************

**Pull request merged**